src/diffusers/loaders.py

```diff
@@ -1283,10 +1284,17 @@ def load_lora_into_text_encoder(cls, state_dict, network_alphas, text_encoder, p
                             f"{name}.out_proj.lora_linear_layer.down.weight"
                         ] = text_encoder_lora_state_dict.pop(f"{name}.to_out_lora.down.weight")
 
-                rank = text_encoder_lora_state_dict[
-                    "text_model.encoder.layers.0.self_attn.out_proj.lora_linear_layer.up.weight"
-                ].shape[1]
+                for name, _ in text_encoder_attn_modules(text_encoder):
+                    rank_key = f"{name}.out_proj.lora_linear_layer.up.weight"
+                    rank.update({rank_key: text_encoder_lora_state_dict[rank_key].shape[1]})
+
                 patch_mlp = any(".mlp." in key for key in text_encoder_lora_state_dict.keys())
+                if patch_mlp:
+                    for name, _ in text_encoder_mlp_modules(text_encoder):
+                        rank_key_fc1 = f"{name}.fc1.lora_linear_layer.up.weight"
+                        rank_key_fc2 = f"{name}.fc2.lora_linear_layer.up.weight"
+                        rank.update({rank_key_fc1: text_encoder_lora_state_dict[rank_key_fc1].shape[1]})
+                        rank.update({rank_key_fc2: text_encoder_lora_state_dict[rank_key_fc2].shape[1]})
```

Comment:
Would it instead be possible to use `_register_state_dict_pre_hook` on LoRALinearLayer so they can look at the incoming weights when the state dict is loaded and change the internal weights to the appropriate shape? This allows us to treat the state dict more transparently and avoid having to construct a rank dict by looking at strings in the passed in state dict.

Reply:
> This allows us to treat the state dict more transparently and avoid having to construct a rank dict by looking at strings in the passed in state dict.

`_register_state_dict_pre_hook` will also need looking at the state dicts if we were to retrieve the ranks no? 

What you're suggesting isn't clear to me. So, need some elaboration.